sensor/common/store/reconciliation/store_test.go

```diff
@@ -0,0 +1,172 @@
+package reconciliation
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	resourceTypeA = "resource_type_A"
+	resourceTypeB = "resource_type_B"
+	fixtureID1    = "1"
+	fixtureID2    = "2"
+)
+
+type storeSuite struct {
+	suite.Suite
+	resourceStore *store
+}
+
+var _ suite.SetupTestSuite = (*storeSuite)(nil)
+
+func (s *storeSuite) SetupTest() {
+	var ok bool
+	s.resourceStore, ok = NewStore().(*store)
+	s.Assert().True(ok)
+}
+
+func Test_StoreSuite(t *testing.T) {
+	suite.Run(t, new(storeSuite))
+}
+
+func (s *storeSuite) Test_Upsert() {
+	testCases := map[string]struct {
+		inputResources    map[string][]string
+		expectedResources map[string][]string
+	}{
+		"One resource": {
+			inputResources: map[string][]string{
+				resourceTypeA: {fixtureID1},
+			},
+			expectedResources: map[string][]string{
+				resourceTypeA: {fixtureID1},
+			},
+		},
+		"Same resource is only added once": {
+			inputResources: map[string][]string{
+				resourceTypeA: {fixtureID1, fixtureID1},
+			},
+			expectedResources: map[string][]string{
+				resourceTypeA: {fixtureID1},
+			},
+		},
+		"Multiple resource types": {
+			inputResources: map[string][]string{
+				resourceTypeA: {fixtureID1, fixtureID2},
+				resourceTypeB: {fixtureID1},
+			},
+			expectedResources: map[string][]string{
+				resourceTypeA: {fixtureID1, fixtureID2},
+				resourceTypeB: {fixtureID1},
+			},
+		},
+	}
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			s.resourceStore.resources = make(map[string]set.StringSet)
```

Comment:
Why not use `initializeStore` here as well?

Reply:
Because `initializeStore` upserts resources in the store. For this test the store needs to be empty.